vbfml/scripts/preprocess_image.py

```diff
@@ -0,0 +1,431 @@
+#!/usr/bin/env python3
+import glob
+import pickle
+import uproot
+import os
+import numpy as np
+import click
+import random
+import pandas as pd
+import matplotlib.pyplot as plt
+import awkward as ak
+
+from math import ceil
+from tqdm import tqdm
+
+from vbfml.training.plot import ImagePlotter
+from vbfml.util import get_process_tag_from_file
+
+pjoin = os.path.join
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option(
+    "-i",
+    "--input-file",
+    required=True,
+    help="Path to the ROOT file.",
+)
+def rotate(: str):
+    """
+    Preprocess the image (in the root file) with a phi-rotation and eta-inversion to have the leading jet in the right center
+    """
+
+    # location and name of new root file -> in a new directory "_preprocessed"
+    output_dir = f"{os.path.dirname(input_file)}_preprocessed_test/"
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+    output_file = output_dir + os.path.basename(input_file)
+
+    # define writable file
+    file = uproot.recreate(output_file)
+
+    # size of the image
+    n_eta_bins: int = 40
+    n_phi_bins: int = 20
+    im_shape = (n_eta_bins, n_phi_bins)
+
+    # load the tree and initilize batch info
+    tree = uproot.open(f"{input_file}:sr_vbf")
+    batch_size = 5000
+    batch_number = ceil(tree.num_entries / batch_size)
+    batch_counter = 0
+
+    # iterate by batches of 500 over the tree, aim to save memoryquit
+    for sub_tree in tree.iterate(
+        ["JetImage_pixels", "leadak4_phi", "leadak4_eta"],
+        step_size=batch_size,
+        library="np",
+    ):
+
+        batch_counter += 1
+        print(f"batch {batch_counter}/{batch_number}")
+
+        batch_index = (batch_counter - 1) * batch_size
+
+        # initialize the new array for the preprocess images
+        New_images = np.zeros((batch_size, n_eta_bins * n_phi_bins), dtype="uint8")
```

Comment:
Could you rename some variables like `New_images` and `New_images_batch` so that they start with lower-case letters? I think it's fine if you just replace the first character.